### ISSUE TITLE
Trouble using/building the ulbridge.so file

### ISSUE THREAD

**Issue by MattSich** [mnottale/uwu#8]
Hello! I'm doing some digging into Ultralight to see if we can use it for a PS5 port of a Unity game but I'm not totally clear about the install step in the readme. I'm curious if you could elaborate a bit? Everything I've tried leads to Unity complaining with a DllNotFoundException so far

**Comment by MattSich** [comment]
ok I'm making progress. I was able to create a ulbridge.dll. I know you said it has to be .so but I think that causes unity trouble because that's a linux target and I'm working on win. 
Unity is loading everything in as expected and I see the ulbridge init log `***ULBRIDGE INIT 0` as expected but Unity crashes right after that. 
Here's the editor log post-crash:
[ulbridge-unity-crash.log](https://github.com/mnottale/uwu/files/12448698/ulbridge-unity-crash.log)

**Comment by MattSich** [comment]
I think the issue may be relate dto cacert.pem and icudt67l.dat files. Curious where those should be located inside the unity project

**Comment by mnottale** [comment]
On my setup they are in a folder named "resources" at the root of the unity project.

**Comment by cspotcode** [comment]
Something that helped me:

I saw `***ULBRIDGE INIT 0` in the Unity log, which indicates the DLL is executing.  Ultralight logs to `ultralight.log`, so I looked at that file for more information about missing resources.

**Comment by borismikhail02** [comment]
Hey Im getting the same issue as MattSich, trying to make a windows desktop game and all all im getting is DllNotFoundException: ulbridge.so assembly:<unknown assembly>... I tried to recompile the soruce to a .dll as i thought Unity might have issues locating a .so (despite your instructions) and that just crashes unity at run. Ive tried putting those "resources" under Assets/Resoruces and ive tried under Assets/StreamingAssets but neither are working

**Comment by mnottale** [comment]
Took another look at things and found an acceptable solution for at least linux, but it should work for windows too (will test that later):

- rename DllImport directive from "ulbridge.so" to "ulbridge", which should load ulbridge.dll on windows and libulbridge.so on linux.
- add instructions to tell unity to preload the dll/so
- added sample compile CLI for linux, the rpath part is important.
Let me know if that helps.

**Comment by mnottale** [comment]
And tested on windows:
- Rebuild the dll using visual studio, C++17 language version, linking with all the ultralight .lib (lib dir)
- Copy the dll renamed ulbridge.dll in assets/plugins, alongside all ultralight dlls (found in sdk bin dir)
- Copy the ultralight bin/resources folder at the toplevel of your project directory.

**Comment by borismikhail02** [comment]
This helped a lot thank you, not getting any errors anymore, and its even logging ****INIT() and Creating texture X Y when isGui is ticked. However nothing is appearing, url is set to default Google. Ive tried to change it to a local file, nothing. If i attach a sprite renderer and use not GUI instead i see a correct sized texture box to appear but its got no texture, invisible box.

**Comment by borismikhail02** [comment]
Yes thank you this helped so much! Got it all working now, I'll let you know if I get anymore issues i get stuck on. You're a life saver !!!

**Comment by borismikhail02** [comment]
Ok this might be a lost hope sadly, I'm starting to think that my use-case is too complex for intergration with unity. I'm trying to intergrate a drag-and-drop block based language via a blockly webtool to Unity, but I thik the drag and drop mouse events might be too complex, as unity cannot recognise them at all, only clicks. Also getting some weird character encoding errors and visual glitches

**Comment by mnottale** [comment]
I think it's doable, my event handling is extremely basic as it's all I needed, but looking at my code for mouse handling it seems it emits a mouseup event to ultralight the next frame after a mouse down, which might explain your dragging issues

**Comment by mnottale** [comment]
Scratch my last comment, ultralight does not support all dom events including dragstart & friends. To implement draggable capabilities one needs to rely on low level mouse events only, which is doable in theory, from what I can read many library do that.
